spec/models/endpoint_spec.rb

```diff
@@ -45,4 +45,16 @@
       end
     end
   end
+
+  context "Uniqueness validation on :url" do
+    it "is not required" do
+      expect(Endpoint.create!).to be_truthy
+      expect(Endpoint.create!).to be_truthy
+    end
+
+    it "raises when provided and already exists" do
+      Endpoint.create!(:url => "abc")
+      expect { Endpoint.create!(:url => "abc") }.to raise_error("Validation failed: Url has already been taken")
+    end
+  end
```

Comment:
Yeah, I was on the fence about writing them since it is just testing rails validations.

My only argument for these kinds of tests is that it shows the intent of the validations for people making future changes.  It's essentially testing this use case.